mapping/field.go

```diff
@@ -77,6 +77,8 @@ type FieldMapping struct {
 	// vector fields.
 	// See: index.DefaultSimilarityMetric & index.SupportedSimilarityMetrics
 	Similarity string `json:"similarity,omitempty"`
+
+	VectorIndexType string `json:"vectorIndexType,omitempty"`
```

Comment:
Snake case for json attributes - you use vectorIndexType here, the document says index_type - converge please.

Reply:
just did that in the next commit :) this is still a WIP PR.